torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -924,13 +924,20 @@ RegisterOperators reg2({
     DEFINE_INT_OP(aten::__lshift__, a << b),
     DEFINE_INT_OP(aten::__rshift__, a >> b),
 
-    DEFINE_GENERIC_BINARY_OP(aten::log, std::log(a) / std::log(b), float),
+    DEFINE_GENERIC_BINARY_OP(
+        aten::log,
+        std::log(a) / std::log(b),
+        float,
+        complex),
     DEFINE_INT_FLOAT_OP(aten::log, std::log(a) / std::log(b), float),
-    DEFINE_SCALAR_SCALAR_BINARY_OP(
+    DEFINE_INT_COMPLEX_OP(aten::log, std::log(a) / std::log(b), complex),
+    DEFINE_FLOAT_COMPLEX_OP(aten::log, std::log(a) / std::log(b), complex),
+    DEFINE_SCALAR_BINARY_OP_AVOID_COLLISION(
         aten::log,
         std::log(a) / std::log(b),
         std::log(a) / std::log(b),
-        float),
+        float,
+        ".Scalar_Scalar"),
```

Comment:
`DEFINE_SCALAR_BINARY_OP_AVOID_COLLISION` is called only ever called in this file with `.Scalar_Scalar` as the final arg; consider adding another macro that accepts the first five arguments and then calls `DEFINE_SCALAR_BINARY_OP_AVOID_COLLISION` with `.Scalar_Scalar` as the sixth.